docs/architecture/ADR-003-borrow-assets.md

```diff
@@ -0,0 +1,121 @@
+# ADR 003: Borrowing assets using uToken collateral
+
+## Changelog
+
+- September 27, 2021: Initial Draft (@toteki)
+
+## Status
+
+Proposed
+
+## Context
+
+One of the base functions of the Umee universal capital facility is to allow users to borrow allowed asset types, using their own uTokens (obtained normally, by depositing assets) as collateral.
+
+## Alternative Approaches
+
+It might be worth considering adding message types which do two or more of the steps below at once (e.g. deposit base asset, convert to uToken collateral, and receive loan all at once). The individual steps to need to be possible in isolation, though.
+
+## Decision
+
+The Cosmos `x/bank` module and the existing `umee/x/leverage` deposit features are prerequisites for these new capabilities.
+
+The flow of events is as follows:
+- Borrower has already used `umee/x/leverage` module to deposit base assets and receive uTokens
+- Borrower deposits uTokens into asset facility as collateral - facility remembers collateral position
+- Borrower requests to borrow assets from capital facility - facility checks request, disburses tokens if acceptable, and remembers borrow position
+- Eventually, borrower pays repayment amount (in full or in part) using borrowed asset denomination
+- Borrower may withdraw collateral uTokens at any time as long as minimum overcollateralization requirement remains satisfied
+- On asset repayment amount = 0 for a given borrow position, facility forgets loan.
+- On uToken collateral amount = 0 for a given collateral position, facility forgets collateral.
+
+Additionally, the following events occur periodically (at the end of every block):
+- Repayment amounts for open borrow positions increases by a borrowed-token-specific interest rate
+- Open borrow positions are checked to see if they fall below liquidation threshold
+
+The `umee/x/leverage` module keeper will be responsible for remembering each open borrow or collateral position.
```

Comment:
What's the difference between a borrow and collateral position? Or are they the same thing?

Reply:
Obsolete now - collateral position was "asset facility is holding 5 of Bob's u/uatom" and borrow position is "bob owes 0.04 ether".